nengo_spinnaker/builder/transmission_parameters.py

```diff
@@ -0,0 +1,401 @@
+        size_in : int
+            Either the size out of the transmitting object (if the function is
+            `None`) or the size out of the function.
+        """
+        super(NodeTransmissionParameters, self).__init__(
+            size_in=size_in,
+            size_out=size_out,
+            transform=transform,
+            slice_out=slice_out
+        )
+        self.pre_slice = pre_slice
+        self.function = function
+
+    def __eq__(self, other):
+        return (super(NodeTransmissionParameters, self).__eq__(other) and
+                self.function is other.function)
+
+    @property
+    def _hashables(self):
+        return super(NodeTransmissionParameters, self)._hashables + (
+            self.function,
+        )
+
+    def __hash__(self):
+        return hash(self._hashables)
+
+    def concats(self, others):
+        """Create new connection parameters which are the result of
+        concatenating this connection with others.
+
+        Parameters
+        ----------
+        others : [NodeTransmissionParameters, ...]
+            Another set of connection parameters to add to the end of this
+            connection.
+
+        Yields
+        ------
+        NodeTransmissionParameters or None
+            Either a new set of transmission parameters, or None if the
+            resulting transform contained no non-zero values.
+        """
+        # Get the outgoing transformation
+        A = self.full_transform(slice_out=False)
+
+        for other in others:
+            # Combine the transforms
+            new_transform = np.dot(other.full_transform(slice_in=False), A)
+
+            # Create a new connection (unless the resulting transform is empty,
+            # in which case don't)
+            if np.any(new_transform != 0):
+                yield NodeTransmissionParameters(
+                    size_in=self.size_in,
+                    size_out=other.size_out,
+                    transform=new_transform,
+                    pre_slice=self.pre_slice,
+                    function=self.function
+                )
+            else:
+                # The transform consisted entirely of zeros so return None.
+                yield None
+
+    @property
+    def as_global_inhibition_connection(self):
+        """Construct a copy of the connection with the optimisation for global
+        inhibition applied.
+        """
+        assert self.supports_global_inhibition
+        transform = self.full_transform(slice_out=False)[0, :]
+        return NodeTransmissionParameters(size_in=self.size_in,
+                                          size_out=1,
+                                          transform=transform,
+                                          pre_slice=self.pre_slice,
+                                          function=self.function)
+
+
+def _get_slice_as_ndarray(sl, size):
+    """Return a slice as a read-only Numpy array."""
+    if isinstance(sl, slice):
+        sl = np.arange(size)[sl]
```

Comment:
This can result in a memory view, which can't be hashed. Should be something like `np.array(np.arange(size)[sl])` or `np.arrange(range(size)[sl])`.